roles/openshift_service_catalog/defaults/main.yml

```diff
@@ -1,6 +1,6 @@
 ---
 openshift_service_catalog_remove: false
-openshift_service_catalog_nodeselector: {"openshift-infra": "apiserver"}
+openshift_service_catalog_nodeselector: "{{ openshift_hosted_infra_selector | default('openshift-infra=apiserver') | map_from_pairs }}"
```

Comment:
Ugh, where'd this `openshift-infra=apiserver` come from? SC seems to be the only place that we're expecting that node selector.

Reply:
Oh well, i think the fix is fine. I'd remove the additional defaults in the tasks though.